doc/developer-guide/generic_type.md

```diff
@@ -0,0 +1,127 @@
+# Generic Type
+
+## Generic Class
+
+### with type arguments
+``` TypeScript
+class A<T> {
+    _x: T;
+
+    constructor(x: T) {
+        this._x = x;
+    }
+    bar<T>(a: T) {
+        console.log('bar: ' + a);
+    }
+}
+```
+
+#### Creation of class type
+
+![](./img/generic_class_type_create.jpg)
+
+The steps to create a generic class type with type arguments are as follows:
+- create class scope  
+  &ensp;The compiler will create a `ClassScope` for each generic class and a `FunctionScope` for each method.
+- create class type  
+  &ensp;Create a generic class type and method types based on `classDeclarationNode`, and associate them with the created `ClassScope` and `FunctionScope`s.
+- create parameters and local variables of method  
+  &ensp;According to `classDeclarationNode`, create parameters and local variables for each method, and save them in the corresponding `FunctionScope`.
+- create the function body of method  
+  &ensp;According to `classDeclarationNode`, create a function body for each method and save it in the corresponding `FunctionScope`.
+
+#### Specialization of class type
+The creation of the specialized class type needs to be based on the `classDeclarationNode` information of the generic class type.
+
+![](./img/generic_class_type_specialize.jpg)
+
+Use `new` to create a new specialized class type:
```

Comment:
Had better record when we will create the specialized class. 

When we meet `NewExpression`, we will know that a new specialized class should be created, which includes new specialized class scope information. 
However, I remind that the new specialized class scope will be created in `semantic tree`? not in `ExpressionProcessor`?

Why put this step into semantic tree? If there any chance that we can done the specialization in frontend?